rust/operator-binary/src/controller.rs

```diff
@@ -0,0 +1,890 @@
+    ApplyRoleGroupConfig {
+        source: stackable_operator::error::Error,
+        rolegroup: RoleGroupRef<TrinoCluster>,
+    },
+    #[snafu(display("failed to apply StatefulSet for {}", rolegroup))]
+    ApplyRoleGroupStatefulSet {
+        source: stackable_operator::error::Error,
+        rolegroup: RoleGroupRef<TrinoCluster>,
+    },
+    #[snafu(display("invalid product config"))]
+    InvalidProductConfig {
+        source: stackable_operator::error::Error,
+    },
+    #[snafu(display("object is missing metadata to build owner reference"))]
+    ObjectMissingMetadataForOwnerRef {
+        source: stackable_operator::error::Error,
+    },
+    #[snafu(display("failed to transform configs"))]
+    ProductConfigTransform {
+        source: stackable_operator::product_config_utils::ConfigError,
+    },
+    #[snafu(display("failed to format runtime properties"))]
+    PropertiesWriteError {
+        source: stackable_operator::product_config::writer::PropertiesWriterError,
+    },
+    #[snafu(display("Failed to load Product Config"))]
+    ProductConfigLoadFailed,
+    #[snafu(display("failed to create rego rules for authorization"))]
+    RegoRuleAuthorizationError { source: authorization::Error },
+    #[snafu(display("failed to get config map [{}/{}]", namespace, name,))]
+    MissingConfigMap {
+        source: stackable_operator::error::Error,
+        name: String,
+        namespace: String,
+    },
+    #[snafu(display(
+        "failed to get [{}] connection string from config map [{}/{}]",
+        product,
+        namespace,
+        name
+    ))]
+    MissingConnectString {
+        product: String,
+        name: String,
+        namespace: String,
+    },
+    #[snafu(display("failed to processing authentication config element from k8s"))]
+    FailedProcessingAuthentication { source: authentication::Error },
+}
+
+type Result<T, E = Error> = std::result::Result<T, E>;
+
+pub async fn reconcile_trino(trino: TrinoCluster, ctx: Context<Ctx>) -> Result<ReconcilerAction> {
+    tracing::info!("Starting reconcile");
+
+    let client = &ctx.get_ref().client;
+    let version = trino_version(&trino)?;
+
+    let validated_config =
+        validated_product_config(&trino, version, &ctx.get_ref().product_config)?;
+
+    // rego rules
+    create_rego_rules(client, &trino)
+        .await
+        .context(RegoRuleAuthorizationError)?;
+
+    let coordinator_role_service = build_coordinator_role_service(&trino)?;
+    client
+        .apply_patch(
+            FIELD_MANAGER_SCOPE,
+            &coordinator_role_service,
+            &coordinator_role_service,
+        )
+        .await
+        .context(ApplyRoleService)?;
+
+    let opa_connect = opa_connect(&trino, &ctx.get_ref().client).await?;
+    let hive_connect = hive_connect(&trino, &ctx.get_ref().client).await?;
+
+    let mut authentication_config = None;
+    if let Some(authentication) = &trino.spec.authentication {
```

Comment:
`authentication_config` doesn't need to be mutable since `if` is an expression

Reply:
compiler disagrees :) ?

Reply:
Looks like this has since been resolved.